pxtlib/tutorial.ts

```diff
@@ -39,7 +39,7 @@ namespace pxt.tutorial {
     function computeBodyMetadata(body: string) {
         // collect code and infer editor
         let editor: string = undefined;
-        const regex = /```(sim|block|blocks|filterblocks|spy|ghost|typescript|ts|js|javascript|template|python)?\s*\n([\s\S]*?)\n```/gmi;
+        const regex = /```(sim|block|blocks|filterblocks|spy|ghost|typescript|ts|js|javascript|template|python)?\s*?\n([\s\S]*?)\n```/gmi;
```

Comment:
```suggestion
        const regex = /```(sim|block|blocks|filterblocks|spy|ghost|typescript|ts|js|javascript|template|python)?\s*?\n([\s\S]*?)\n?```/gmi;
```
seems sufficient to handle the empty case, but allows the last line of code to end in ```, which isn't that big of a deal but ¯\\_(ツ)_/¯

we decided to go with a fix that just doesn't emit empty comments for now, as this would need a LOT of testing / validation